controllers/common/ansible.go

```diff
@@ -5,26 +5,111 @@ package common
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"reflect"
 	"regexp"
+	"sort"
+	"strconv"
 	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
+	ctrl "sigs.k8s.io/controller-runtime"
 
 	policyv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
 )
 
+const (
+	ControllerName             string = "policy-automation"
+	PolicyAutomationLabel      string = "ansiblejob.policy.open-cluster-management.io/policyautomation-name"
+	PolicyAutomationGeneration string = "policy.open-cluster-management.io/policyautomation-generation"
+	AnsiblejobIndex            string = "policy.open-cluster-management.io/ansiblejob-index"
+)
+
+var log = ctrl.Log.WithName(ControllerName)
+
+var ansibleJobRes = schema.GroupVersionResource{
+	Group: "tower.ansible.com", Version: "v1alpha1",
+	Resource: "ansiblejobs",
+}
+
+// Get Ansiblejob order among same policyautomation
+func GetAnsibleJobOrder(policyAutomation *policyv1beta1.PolicyAutomation,
+	dynamicClient dynamic.Interface,
+) int {
+	ansiblejobList, err := dynamicClient.Resource(ansibleJobRes).
+		Namespace(policyAutomation.GetNamespace()).List(
+		context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", PolicyAutomationLabel, policyAutomation.GetName()),
+		},
+	)
+	if err != nil {
+		return 0
+	}
+
+	return len(ansiblejobList.Items) + 1
+}
+
+// Sort and get policy automation's generation of the last ansiblejob
+func GetLastAjGeneration(policyAutomation *policyv1beta1.PolicyAutomation,
+	dynamicClient dynamic.Interface,
+) (string, error) {
+	ansiblejobList, err := dynamicClient.Resource(ansibleJobRes).Namespace(policyAutomation.GetNamespace()).List(
+		context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", PolicyAutomationLabel, policyAutomation.GetName()),
+		},
+	)
+
+	ansiblejobLen := len(ansiblejobList.Items)
+
+	if err != nil {
+		log.Error(err, "Faile to get ansiblejob list")
+
+		return "0", err
+	}
+
+	if ansiblejobLen == 0 {
+		return "0", nil
+	}
```

Comment:
Mainly a spelling error ("Faile"), but I think reordering these makes sense also.
```suggestion
	if err != nil {
		log.Error(err, "Failed to get ansiblejob list")

		return "0", err
	}

	ansiblejobLen := len(ansiblejobList.Items)
	if ansiblejobLen == 0 {
		return "0", nil
	}
```